plugin/evm/message/message.go

```diff
@@ -22,8 +22,14 @@ const (
 )
 
 var (
-	_ GossipMessage = AtomicTxGossip{}
-	_ GossipMessage = EthTxsGossip{}
+	_ GossipMessage = AtomicTxGossip{} // TODO deprecate
+	_ GossipMessage = EthTxsGossip{}   // TODO deprecate
+	_ GossipMessage = (*PullAtomicTxsGossip)(nil)
+	_ GossipMessage = (*AtomicTxsGossip)(nil)
+	_ GossipMessage = (*PullEthTxsGossip)(nil)
+	// TODO rename after EthTxsGossip is deprecated
+	_ GossipMessage = (*EthTxsGossipV2)(nil)
```

Comment:
This is done purposefully so that when we are migrating to this new gossip implementation the old gossip can exist in parallel. Deprecation will be simple since we just have to delete the old message type.